clodius/tiles/bigbed.py

```diff
@@ -0,0 +1,407 @@
+        with the `|cos:id` tag in the tile id
+    chromsizes: [[chrom, size],...]
+        A 2d array containing chromosome names and sizes. Overrides the
+        chromsizes in chromsizes_map
+
+    Returns
+    -------
+    tile_list: [(tile_id, tile_data),...]
+        A list of tile_id, tile_data tuples
+    '''
+    TILE_SIZE = 1024
+    min_elements = -1
+    max_elements = -1
+    
+    generated_tiles = []
+    for tile_id in tile_ids:
+        tile_option_parts = tile_id.split('|')[1:]
+        tile_no_options = tile_id.split('|')[0]
+        tile_id_parts = tile_no_options.split('.')
+        tile_position = list(map(int, tile_id_parts[1:3]))
+        return_value = tile_id_parts[3] if len(tile_id_parts) > 3 else 'mean'
+
+        range_mode = return_value if return_value in range_modes else None
+
+        tile_options = dict([o.split(':') for o in tile_option_parts])
+
+        if 'min' in tile_options:
+            min_elements = int(tile_options['min'])
+        if 'max' in tile_options:
+            max_elements = int(tile_options['max'])
+            
+        if min_elements > max_elements:
+            temp_min_elements = min_elements
+            min_elements = max_elements
+            max_elements = temp_min_elements 
+        elif min_elements == max_elements:
+            min_elements = max_elements
+            max_elements = min_elements + 1
+            
+        if max_elements <= 0:
+            min_elements = 1
+            max_elements = 10
+
+        if chromsizes:
+            chromnames = [c[0] for c in chromsizes]
+            chromlengths = [int(c[1]) for c in chromsizes]
+            chromsizes_to_use = pd.Series(chromlengths, index=chromnames)
+        else:
+            chromsizes_id = None
+            if 'cos' in tile_options:
+                chromsizes_id = tile_options['cos']
+            if chromsizes_id in chromsizes_map:
+                chromsizes_to_use = chromsizes_map[chromsizes_id]
+            else:
+                chromsizes_to_use = None
+
+        zoom_level = tile_position[0]
+        tile_pos = tile_position[1]
+
+        # this doesn't combine multiple consequetive ids, which
+        # would speed things up
+        if chromsizes_to_use is None:
+            chromsizes_to_use = get_chromsizes(bbpath)
+
+        max_depth = get_quadtree_depth(chromsizes_to_use)
+        tile_size = TILE_SIZE * 2 ** (max_depth - zoom_level)
+        start_pos = tile_pos * tile_size
+        end_pos = start_pos + tile_size
+    
+        tile_value = get_bigbed_tile(
+            bbpath,
+            zoom_level,
+            start_pos,
+            end_pos,
+            chromsizes_to_use,
+            range_mode=range_mode,
+            min_elements=min_elements,
+            max_elements=max_elements
+        )
+
+        generated_tiles += [(tile_id, tile_value)]
+    
+    return generated_tiles
+
+
+def chromsizes(filename):
```

Comment:
Just reroute to the bigWig function, no?

Reply:
I added another commit to call the bigwig code for `get_chromsizes`.